### PR TITLE
[FEATURE] generateThemeDesignerResources task: Create css_variables.less

### PR DESCRIPTION
**Pull request by flovogt**
[FEATURE] Enhance task generateThemeDesignerResources.js to create css_variables.less files.
BLI: CPOUI5FOUNDATION-487

### CODE REVIEW

**Comment by flovogt** [comment]
Line https://github.com/SAP/ui5-builder/pull/730/files#diff-92c6913aec2b1a1b12c249189bafc7a0abbe5ccee607f72e847023916ed58796L59 and the following lines are creating a complete new `.theming` file. So, the existing theming file is ignored is this desired?

**Comment by matz3** [comment]
Yes, the `.theming` file is either generated or copied from the `sap/ui/core` namespace. If it exists in the sources it will be overwritten.

File: lib/tasks/generateThemeDesignerResources.js [hunk 1]
```diff
@@ -80,6 +153,39 @@ async function generateThemeDotTheming({workspace, combo, themeFolder}) {
...
+			cssVariablesSourceLessResource.map((cssVariableSourceLess) => {
+				const themeFolder = posixPath.dirname(cssVariableSourceLess.getPath());
+				log.verbose(`Generating css_variables.less for theme ${themeFolder}`);
+				return createCssVariablesLessResource({
+					workspace, combo, themeFolder
+				});
+			})
+		);
+
+		return cssVariablesLessResources.map(async (resource) => {
```
Comment (matz3) [inline lib/tasks/generateThemeDesignerResources.js:181]: Isn't `Promise.all` missing here?

**Review by matz3 — CHANGES_REQUESTED** [review]

File: lib/tasks/generateThemeDesignerResources.js (hunk 1 above)
Comment (flovogt) [reply lib/tasks/generateThemeDesignerResources.js:181]: Agreed on moving write logic to loop above

Comment (flovogt) [reply lib/tasks/generateThemeDesignerResources.js:187]: Done

**Review by matz3 — APPROVED** [review]

**Pull request merged**